src/lib/ndpi_main.c

```diff
@@ -2475,6 +2479,52 @@ u_int16_t ndpi_network_port_ptree_match(struct ndpi_detection_module_struct *ndp
 
 /* ******************************************* */
 
+u_int16_t ndpi_network_port_ptree6_match(struct ndpi_detection_module_struct *ndpi_str,
+					 struct in6_addr *pin,
+					 u_int16_t port /* network byte order */)
+{
+  ndpi_prefix_t prefix;
+  ndpi_patricia_node_t *node;
+
+  if(!ndpi_str || !ndpi_str->protocols_ptree6)
+    return(NDPI_PROTOCOL_UNKNOWN);
+
+  /* TODO: check on "private" addresses? */
```

Comment:
Makes sense for me. There could be also risk set, if that happens.

Reply:
More than an anomaly, it is an optimization. Like in the ipv4 case, if there are not link local/"private" addresses in the tree (and, by default, there are not) we can avoid the lookup if the flow addressees are of these types